File: web/types/support-request.ts
export type RequestStatus = "Open" | "InProgress" | "Completed";
export type RequestPriority = "Low" | "Medium" | "High";

export interface SupportRequest {
  id: number;
  title: string;
  description: string;
  status: RequestStatus;
  priority: RequestPriority;
  createdAt: string;
}

export interface SupportRequestInput {
  title: string;
  description: string;
  status: RequestStatus;
  priority: RequestPriority;
}

export interface DashboardStats {
  open: number;
  inProgress: number;
  completed: number;
}

File: api/Models/SupportRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SupportFlow.Api.Models;

public class SupportRequest
{
    public int Id { get; set; }

    [MaxLength(120)]
    public required string Title { get; set; }

    [MaxLength(1000)]
    public required string Description { get; set; }

    public RequestStatus Status { get; set; }

    public RequestPriority Priority { get; set; }

    public DateTime CreatedAt { get; set; }
}
